### PR TITLE
Get rid of ResourceWarnings produced by the tests

### PR DESCRIPTION
**Pull request by gsnedders**
This gets us to a point of being able to treat warnings as errors, and gets rid of all ResourceWarnings.

### CODE REVIEW

**Comment by gsnedders** [comment]
Okay, the last commit here was a bit optimistic. Let's drop it from here for now. 🙃

**Comment by gsnedders** [comment]
Not sure why the `resources/` tests failed, looks like something is up with Firefox, let's just try again?

File: tools/wptserve/wptserve/response.py [hunk 1]
```diff
@@ -203,10 +203,12 @@ def iter_content(self, read_file=False):
         elif isinstance(self.content, str):
             yield self.content.encode(self.encoding)
         elif hasattr(self.content, "read"):
-            if read_file:
-                yield self.content.read()
-            else:
-                yield self.content
+            while True:
+                read = self.content.read(64 * 1024)
```
Comment (jgraham) [inline tools/wptserve/wptserve/response.py:207]: I think this deserves a comment. Why are we manually chunking into 64kB chunks when reading?

**Review by jgraham — CHANGES_REQUESTED** [review]

File: tools/wptserve/wptserve/response.py (hunk 1 above)
Comment (gsnedders) [reply tools/wptserve/wptserve/response.py:207]: This was meant to match the behaviour of `ResponseWriter.write_content_file`, except that seems to be 32KiB chunks. Why? Presumably to just avoid ending up with potentially vast files in memory.

Comment (jgraham) [reply tools/wptserve/wptserve/response.py:207]: OK, can you add that as a comment? :)

Comment (gsnedders) [reply tools/wptserve/wptserve/response.py]: Done!

**Comment by gsnedders** [comment]
#39660 should, I believe, fix the current CI failures on TC.

**Review by jgraham — APPROVED** [review]

**Pull request merged**